spec/facility_spec.rb

```diff
@@ -23,4 +31,101 @@
       expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
     end
   end
+
+  describe '#collect fees' do
+    it 'can collect fees' do
+      @facility.collect_fee(@cruz)
+      expect(@facility.collected_fees).to eq(100)
+    end
+
+    it 'can collect fees multiple times' do
+      @facility.collect_fee(@cruz)
+      expect(@facility.collected_fees).to eq(100)
+      @facility.collect_fee(@camaro)
+      expect(@facility.collected_fees).to eq(125)
+      @facility.collect_fee(@bolt)
+      expect(@facility.collected_fees).to eq(325)
+    end
+  end
+
+  describe '#register vehicle' do
+    it 'can register a vehicle' do
+      @facility.add_service("Vehicle Registration")
+      @facility.register_vehicle(@cruz)
+      expect(@cruz.registration_date).to be_an_instance_of(Date)
+      expect(@facility.registered_vehicles.count).to eq(1)
+      @facility.register_vehicle(@bolt)
+      expect(@bolt.registration_date).to be_an_instance_of(Date)
+      @facility.register_vehicle(@camaro)
+      expect(@camaro.registration_date).to be_an_instance_of(Date)
+      expect(@facility.registered_vehicles.count).to eq(3)
+    end
+
+    it 'can fail to register a vehicle' do
+      expect(@facility.register_vehicle(@cruz)).to eq false
+    end
+
+    it 'sets a plate type' do
+      @facility.add_service("Vehicle Registration")
+      @facility.register_vehicle(@cruz)
+      expect(@cruz.plate_type).to eq(:regular)
+      @facility.register_vehicle(@bolt)
+      expect(@bolt.plate_type).to eq(:ev)
+      @facility.register_vehicle(@camaro)
+      expect(@camaro.plate_type).to eq(:antique)
+
+    end
+  end
+
+  describe '#administer written test' do
+    it 'can give a written a test' do
+      @facility.administer_written_test(@registrant_1)
+      expect(@registrant_1.license_data[:written]).to eq false
+      @facility.add_service('Written Test')
+      @facility.administer_written_test(@registrant_1)
+      expect(@registrant_1.license_data[:written]).to eq true
```

Comment:
Good side effect testing, consider also testing the services array has been updated here for extra testing